src/components/Network.js

```diff
@@ -33,27 +34,45 @@ class Network extends React.Component {
       connected_peers: [],
       known_peers: [],
       loaded: false,
-      isLoading: false
+      isLoading: false,
+      peers: [],
+      peer_id: '',
     }
 
     this.loadData = this.loadData.bind(this);
   }
 
   componentDidMount() {
-    this.loadData();
-    this.loadTimer = setInterval(this.loadData, 1000);
+    const { match: { params } } = this.props;
+
+    networkApi.getPeers().then((peers) => {
+      this.setState({peers}, () => {
+        let peer_id = params.peer_id;
+        if (!peer_id) {
+          peer_id = peers[0];
+        }
+        this.props.history.push(`/network/${this.state.peers[0]}`);
+        this.setState({peer_id}, () => {
+          this.loadData();
+          this.loadTimer = setInterval(this.loadData, 1000);
+        });
+      });
+    });
+  }
+
+  onPeerChange(event) {
+    const peer_id = event.target.value;
+    this.props.history.push(`/network/${peer_id}`);
+    this.setState({peer_id}, () => { this.loadData() });
   }
 
   loadData() {
     if (this.state.isLoading) return;
 
     this.setState({ isLoading: true }, () => {
-      networkApi.getPeers().then((peers) => {
+      networkApi.getPeer(this.state.peer_id).then((peer) => {
```

Comment:
What if this request fails?

Reply:
We don't have a good way to handle, or show errors in general for Explorer yet. With the way we use Axios nowadays, it's even more tricky. I'll open an issue for that.

For this case It will retry every second and show the Loading component while it's not succeeded